File: niuniu-take-out-springboot3/server/src/main/java/common/niuniu/service/CategoryService.java
package common.niuniu.service;

import common.niuniu.dto.CategoryDTO;
import common.niuniu.dto.CategoryTypePageDTO;
import common.niuniu.entity.Category;
import common.niuniu.result.PageResult;

import java.util.List;

public interface CategoryService {
    void addCategory(CategoryDTO categoryDTO);

    PageResult getPageList(CategoryTypePageDTO categoryTypePageDTO);

    List<Category> getList(Integer type);

    Category getById(Integer id);
    void onOff(Integer id);

    void udpate(CategoryDTO categoryDTO);

    void delete(Integer id);

}

File: niuniu-take-out-kotlin/server/src/main/kotlin/common/niuniu/service/CategoryService.kt
package common.niuniu.service

import common.niuniu.dto.CategoryDTO
import common.niuniu.dto.CategoryTypePageDTO
import common.niuniu.po.Category
import common.niuniu.result.PageResult

interface CategoryService {
    fun addCategory(categoryDTO: CategoryDTO?)

    fun getPageList(categoryTypePageDTO: CategoryTypePageDTO?): PageResult?

    fun getList(type: Int): List<Category>

    fun getById(id: Int?): Category?
    fun onOff(id: Int?)

    fun udpate(categoryDTO: CategoryDTO?)

    fun delete(id: Int?)
}
